File: PERSISTENCIA/EJERCICIO-3/PYTHON/ArchivoCliente.py
import json
import os

class Cliente:
    def __init__(self, id, nombre, telefono):
        self.id = id
        self.nombre = nombre
        self.telefono = telefono

    def to_dict(self):
        return {
            "id": self.id,
            "nombre": self.nombre,
            "telefono": self.telefono
        }

    @staticmethod
    def from_dict(data):
        return Cliente(data["id"], data["nombre"], data["telefono"])

class ArchivoCliente:
    def __init__(self, nomA):
        self.nomA = nomA + ".json"

    def crear_archivo(self):
        if not os.path.exists(self.nomA):
            with open(self.nomA, 'w') as f:
                json.dump([], f)

    def guardar_cliente(self, cliente):
        with open(self.nomA, 'r') as f:
            datos = json.load(f)
        datos.append(cliente.to_dict())
        with open(self.nomA, 'w') as f:
            json.dump(datos, f)

    def buscar_cliente(self, c):
        with open(self.nomA, 'r') as f:
            datos = json.load(f)
        for item in datos:
            if item["id"] == c:
                return Cliente.from_dict(item)
        return None

    def buscar_celular_cliente(self, c):
        cliente = self.buscar_cliente(c)
        if cliente:
            return f"Nombre: {cliente.nombre}, Celular: {cliente.telefono}"
        return "Cliente no encontrado"

c1 = Cliente(1, "Ana Pérez", 77712345)
c2 = Cliente(2, "Luis García", 72145678)
c3 = Cliente(3, "Marta Rojas", 76543210)

archivo = ArchivoCliente("clientes")
archivo.crear_archivo()
archivo.guardar_cliente(c1)
archivo.guardar_cliente(c2)
archivo.guardar_cliente(c3)

res = archivo.buscar_cliente(2)
if res:
    print(res.nombre, res.telefono)
else:
    print("No encontrado")

print(archivo.buscar_celular_cliente(3))

File: PERSISTENCIA/EJERCICIO-3/JAVA/ArchivoCliente.java
import java.io.*;

public class ArchivoCliente {
    private String nomA;
    private File archivo;

    public ArchivoCliente(String n) {
        this.nomA = n;
    }

    public void crearArchivo() {
        archivo = new File(nomA + ".txt");
        try {
            if (archivo.createNewFile()) {
                System.out.println("Archivo creado con éxito.");
            } else {
                System.out.println("El archivo ya existe.");
            }
        } catch (IOException ex) {
            ex.printStackTrace(System.out);
        }
    }

    public void guardaCliente(Cliente c) {
        try (FileWriter fw = new FileWriter(archivo, true)) {
            fw.write(c.getId() + "\n");
            fw.write(c.getNombre() + "\n");
            fw.write(c.getTelefono() + "\n");
        } catch (IOException e) {
            e.printStackTrace(System.out);
        }
    }

   
    public Cliente buscarCliente(int c) {
        try (BufferedReader br = new BufferedReader(new FileReader(archivo))) {
            String linea;
            while ((linea = br.readLine()) != null) {
                int id = Integer.parseInt(linea);
                String nombre = br.readLine();
                int telefono = Integer.parseInt(br.readLine());

                if (id == c) {
                    return new Cliente(id, nombre, telefono);
                }
            }
        } catch (IOException ex) {
            ex.printStackTrace(System.out);
        }
        return null;
    }

  
    public void buscarCelularCliente(int c) {
        Cliente encontrado = buscarCliente(c);
        if (encontrado != null) {
            System.out.println("Cliente encontrado:");
            System.out.println(encontrado.toString());
        } else {
            System.out.println("Cliente no encontrado.");
        }
    }

    public static void main(String[] args) {
        ArchivoCliente arch = new ArchivoCliente("clientes");
        arch.crearArchivo();

        Cliente cli1 = new Cliente(1, "Ana Torres", 5551234);
        Cliente cli2 = new Cliente(2, "Luis Gómez", 5555678);
        Cliente cli3 = new Cliente(3, "Sofía Pérez", 5559012);

        arch.guardaCliente(cli1);
        arch.guardaCliente(cli2);
        arch.guardaCliente(cli3);

        System.out.println("\nBuscando cliente con ID 2:");
        Cliente buscado = arch.buscarCliente(2);
        if (buscado != null) {
            System.out.println(buscado);
        }

        System.out.println("\nBuscando celular de cliente con ID 3:");
        arch.buscarCelularCliente(3);
    }
}
